xdsl/parser/attribute_parser.py

```diff
@@ -400,7 +410,7 @@ def _parse_unregistered_attr_body(self, start_pos: Position | None) -> str:
         assert body is not None
         return body
 
-    def _parse_optional_builtin_parametrized_type(self) -> ParametrizedAttribute | None:
```

Comment:
nothing was actually relying on this being a ParametrizedAttribute, and there's no `A & B` intersection type mechanism